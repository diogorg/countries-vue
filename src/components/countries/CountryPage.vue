<template>
    <CountryHeader />
    <div class="main q-pb-md">
        <CountryForm />
        <CountryResult />
    </div>
</template>

<script>
import CountryHeader from './CountryHeader.vue'
import CountryForm from './CountryForm.vue'
import CountryResult from './CountryResult.vue'
export default {
    name: 'CountryPage',
    components: {
        CountryHeader,
        CountryForm,
        CountryResult
    },
    data: () => ({}),
    beforeCreate() {
        const quasar = this.$q
        this.$store.commit('getAllCountriesAndRegions', { quasar })
    },
    setup() { }
}
</script>

<style scoped>
.main {
    height: 100%;
    width: 100%;
    display: flex;
    justify-content: space-between;
    vertical-align: top;
    flex-direction: row;
    align-items: stretch;
    overflow: auto;
    align-items: stretch;
}
</style>