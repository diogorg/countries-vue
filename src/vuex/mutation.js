import {getCountriesFromCache, getCountriesFromApi} from './model/get-countries'
import {apiFailNotification, apiSuccessNotification} from '@/notifications'
import {filterCountries} from './model/filter-countries'

export const mutation = {
    selectCountry(state, { country }) {
        state.dialog = true
        state.selected = country
    },
    closeDialog(state) {
        state.dialog = false
        state.selected = null
    },
    filterCountries(state, { name, region, lang }) {
        state.countries = filterCountries(state.total, name, region, lang)
    },
    getAllCountriesAndRegions(state, { quasar }) {
        const data = getCountriesFromCache() || getCountriesFromApi()
        if (!data) {
            apiFailNotification(quasar)
            return
        }
        apiSuccessNotification(quasar)
        state.countries = data.countries
        state.total = data.total
        state.regions = data.regions
        state.langs = data.langs
        return
    }
}