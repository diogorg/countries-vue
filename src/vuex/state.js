export const state = {
    total: [],
    countries: [],
    regions: ['ALL'],
    langs: ['ALL'],
    dialog: false,
    selected: null
}