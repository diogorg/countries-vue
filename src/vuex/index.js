import Vuex from 'vuex'
import {state} from './state'
import {mutation} from './mutation'

export const store = new Vuex.Store({
    state: () => state,
    mutations: mutation,
    actions: {}
})