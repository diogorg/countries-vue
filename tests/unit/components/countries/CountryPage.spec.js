import { shallowMount } from '@vue/test-utils'
import CountryPage from '@/components/countries/CountryPage.vue'
import { createStore } from 'vuex'
import { state } from '@/vuex/state'
import { components } from './basic-components'

const store = createStore({
    state() {
        return state
    },
    mutations: {
        getAllCountriesAndRegions() {
            return []
        }
    }
})

const global = {
    components,
    mocks: {
        $store: store
    }
}

describe('CountryPage.vue', () => {
    it('Should match snapshot', () => {
        const wrap = shallowMount(CountryPage, {
            global
        })
        expect(wrap.html()).toMatchSnapshot()
    })
})