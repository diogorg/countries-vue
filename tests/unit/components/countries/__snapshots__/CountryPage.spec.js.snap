// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryPage.vue Should match snapshot 1`] = `
<country-header-stub></country-header-stub>
<div class="main q-pb-md">
  <country-form-stub></country-form-stub>
  <country-result-stub></country-result-stub>
</div>
`;
